### PR TITLE
#5239 Add it with restart workspace more case

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 1]
```diff
@@ -0,0 +1,1599 @@
...
+    // create a node
+    cy.findByTitle(/create node/i).click();
+    cy.get('input[name=hostname]').type(hostname);
+    cy.get('input[name=port]').type(generate.port().toString());
+    cy.get('input[name=user]').type(generate.userName());
+    cy.get('input[name=password]').type(generate.password());
+    cy.findByText('CREATE').click();
+    cy.findByTestId('nodes-dialog-close-button').click();
+
+    // add new added node into workspace
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:56]: _newly_

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 2]
```diff
@@ -0,0 +1,1599 @@
...
+    });
+
+    cy.findAllByRole('dialog')
+      .filter(':visible')
+      .should('have.length', 1)
+      .within(() => {
+        cy.get('table')
+          .should('have.length', 1)
+          .within(($table) => {
+            // checked the new added hostname
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:78]: _check ... newly__

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 3]
```diff
@@ -0,0 +1,1599 @@
...
+    cy.findByText('CREATE').click();
+    cy.findByTestId('nodes-dialog-close-button').click();
+
+    // add new added node into workspace
+    cy.switchSettingSection(SETTING_SECTIONS.nodes);
+    cy.get('div.section-page-content').within(() => {
+      cy.findByTitle('Add Node').click();
+    });
+
+    // re-open select node
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:62]: _reopen_

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 4]
```diff
@@ -0,0 +1,1599 @@
...
+      });
+
+    cy.get('div.section-page-content').within(() => {
+      cy.findByText(hostname)
+        .should('exist')
+        .siblings('td')
+        // the "Used" column
+        .eq(3)
+        .invoke('html')
+        // there is no service assign to this node yet
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:97]: _assigned_

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 5]
```diff
@@ -0,0 +1,1599 @@
...
+    cy.findAllByRole('dialog')
+      .filter(':visible')
+      .should('have.length', 1)
+      .within(() => {
+        cy.get('table')
+          .should('have.length', 1)
+          .within(($table) => {
+            // assert the Services should be 0
+            cy.getTableCellByColumn($table, 'Services', '0').should('exist');
+            // checked the new added hostname
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:118]: _check ... newly_

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 6]
```diff
@@ -0,0 +1,1599 @@
...
+  it('should able to restart broker by directly click button', () => {
+    cy.switchSettingSection(SETTING_SECTIONS.dangerZone, 'Restart this broker');
+    cy.findAllByRole('dialog')
+      .filter(':visible')
+      .should('have.length', 1)
+      .within(() => {
+        cy.findByText('CANCEL').click();
+      });
+
+    // re-click again
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:531]: _click_

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 7]
```diff
@@ -0,0 +1,1599 @@
...
+    cy.findByText('Start topic', { exact: false })
+      .next()
+      .should('have.text', '[OK]');
+    cy.findByText('Start worker', { exact: false })
+      .next()
+      .should('have.text', '[OK]');
+    cy.findAllByText('Restart workspace', { exact: false })
+      .eq(2)
+      .next()
+      .should('have.text', '[OK]');
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:675]: Maybe we could assert if there's any error or not?

**Comment by wu87988622** [comment]
retry manager-it

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts [hunk 8]
```diff
@@ -0,0 +1,1599 @@
...
+        cy.findByText('RESTART').click();
+      });
+
+    cy.findByText('ERROR', { exact: false });
+
+    cy.request('PUT', 'api/workers/workspace1/start?group=worker');
+
+    cy.route({
+      method: 'PUT',
+      url: new RegExp('.api.workers.*start.*'),
```
Comment (eechih) [inline ohara-manager/client/cypress/it/navigator/dangerZone.ts:1496]: 這樣的表示法有點難讀 :( ，可以試試看 `api/workers/*/start**`

Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:1496]: 好的
我在調整

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:56]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:78]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:62]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:97]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 5 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:118]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 6 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:531]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.ts (hunk 7 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.ts:675]: 會調整為
第一隻檢查會詳細的檢查
後面會檢查比較指標性的訊息
感謝

**Pull request merged**